src/utils/PaymentsConverter.ts

```diff
@@ -0,0 +1,236 @@
+  codiceContestoPagamento: CodiceContestoPagamento
+): Either<Error, PaymentsCheckResponse> {
+  if (
+    iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.esito ===
+    PPTPortTypes.Esito.KO
+  ) {
+    return left(new Error(ControllerError.REQUEST_REJECTED));
+  }
+  const errorOrPaymentCheckResponse = PaymentsCheckResponse.decode({
+    importoSingoloVersamento:
+      iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+        .importoSingoloVersamento,
+    codiceContestoPagamento,
+    ibanAccredito:
+      iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+        .ibanAccredito,
+    causaleVersamento:
+      iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+        .causaleVersamento,
+    enteBeneficiario: {
+      codiceIdentificativoUnivoco:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.identificativoUnivocoBeneficiario,
+      denominazioneBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.denominazioneBeneficiario,
+      codiceUnitOperBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.codiceUnitOperBeneficiario,
+      denomUnitOperBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.denomUnitOperBeneficiario,
+      indirizzoBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.indirizzoBeneficiario,
+      civicoBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.civicoBeneficiario,
+      capBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.capBeneficiario,
+      localitaBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.localitaBeneficiario,
+      provinciaBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.provinciaBeneficiario,
+      nazioneBeneficiario:
+        iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+          .enteBeneficiario.nazioneBeneficiario
+    },
+    spezzoniCausaleVersamento:
+      iNodoVerificaRPTOutput.nodoVerificaRPTRisposta.datiPagamentoPA
+        .spezzoniCausaleVersamento
+  });
+
+  if (errorOrPaymentCheckResponse.isLeft()) {
+    return left(new Error(ControllerError.ERROR_INVALID_INPUT));
+  }
+  return right(errorOrPaymentCheckResponse.value);
+}
+
+// Convert PaymentsActivationRequest (controller) to PaymentsActivationRequestPagoPa (PagoPa API)
+export function getPaymentsActivationRequestPagoPa(
+  pagoPaConfig: PagoPaConfig,
+  paymentsActivationRequest: PaymentsActivationRequest
+): Either<Error, InodoAttivaRPTInput> {
+  return right({
```

Comment:
return the plain object not an either

Reply:
It's done in this way because when pagopaapi typed structures will be available, data will be validated using decode (so, we'll have right and left options).

Reply:
that's right. then please add comments wherever something is TBD later (to help reviewers). btw, if this method it's just an alias for mapLeft() it's better to avoid it entirely and just use 
InodoAttivaRPTInput.mapLeft(new Error()) in the caller